### DISCUSSION TITLE
Return the matching search value in a multi-string search (i.e. IndexOfAny with SearchValues<string>)

### DISCUSSION THREAD

**Discussion by f2bo** [dotnet/runtime#110663] [Ideas] [answered]
.NET 9 introduced multi-string search with `IndexOfAny(this ReadOnlySpan<char> span, SearchValues<string> values)`, which is great if you need to find a set of matching strings. However, and I would say more often than not, in addition to the location, you also need to know which of the search values was found, for example, to count instances for each of the search values or even just to advance a pointer past the match and continue processing.

Until now, when searching for single characters, this was trivial since the matching search value had length 1 and could be directly retrieved from the location returned by `IndexOfAny`. When searching for strings, however, it now becomes necessary to execute an additional comparison at the match location to determine which of the search values was found. 

I'm aware that you could get this information by using a `Regex` instead, but it seems odd given that `IndexOfAny` has already determined the matching search value. Wouldn't it be possible to also return it, even if it's in the form of an index into the `ReadOnlySpan<string>` that was originally used to create the `SearchValues` instance. This could be provided either as an additional output parameter or by returning a tuple that also includes the location.

**Comment by MihaZupan** [comment, marked as answer]
Moved this to an issue: #110670
